pkg/controller/v1alpha1/localmodel/controller.go

```diff
@@ -157,18 +158,26 @@ func (c *LocalModelReconciler) createPVC(ctx context.Context, spec v1.Persistent
 }
 
 // ReconcileForIsvcs Get all isvcs with model cache enabled, create pvs and pvcs, remove pvs and pvcs in namespaces without isvcs.
-func (c *LocalModelReconciler) ReconcileForIsvcs(ctx context.Context, localModel *v1alpha1api.LocalModelCache, nodeGroup *v1alpha1api.LocalModelNodeGroup, jobNamespace string) error {
+func (c *LocalModelReconciler) ReconcileForIsvcs(ctx context.Context, localModel *v1alpha1api.LocalModelCache, localModelNodeGroups map[string]*v1alpha1api.LocalModelNodeGroup, jobNamespace string) error {
 	isvcs := &v1beta1.InferenceServiceList{}
 	if err := c.Client.List(ctx, isvcs, client.MatchingFields{localModelKey: localModel.Name}); err != nil {
 		c.Log.Error(err, "List isvc error")
 		return err
 	}
 	isvcNames := []v1alpha1.NamespacedName{}
-	// namespaces with isvcs deployed
-	namespaces := make(map[string]struct{})
+	// namespaces with isvcs deployed and their node groups
+	namespaceToNodeGroups := make(map[string]map[*v1alpha1api.LocalModelNodeGroup]struct{})
 	for _, isvc := range isvcs.Items {
 		isvcNames = append(isvcNames, v1alpha1.NamespacedName{Name: isvc.Name, Namespace: isvc.Namespace})
-		namespaces[isvc.Namespace] = struct{}{}
+		if isvcNodeGroup, ok := isvc.ObjectMeta.Annotations[constants.NodeGroupAnnotationKey]; ok {
+			if nodeGroup, ok := localModelNodeGroups[isvcNodeGroup]; ok {
+				namespaceToNodeGroups[isvc.Namespace][nodeGroup] = struct{}{}
+			} else {
+				c.Log.Info("Didn't find isvc node group in model cache node groups", "isvc name", isvc.Name, "isvc node group", isvcNodeGroup, "model cache node groups", maps.Keys(localModelNodeGroups))
```

Comment:
Should we add an inference service validating webhook to check if the nodegroup annotation is valid? cc @yuzisun

Reply:
Step 1: Model is cached on nodegroup1 and nodegroup2 and user deploy an isvc on nodegroup1.
Step 2: Admin updates the model cache to only cache on nodegroup1.
Should this be rejected?

Reply:
Step 1: Model is cached on nodegroup1 and nodegroup2.
Step 2: Admin updates the model cache to only cache on nodegroup1.
The model should be removed from the nodegroup2. Can you check if it is removed?
Should we delete pv & pvc for nodegroup2?